lib/studies/reply/inbox.dart

```diff
@@ -278,31 +290,342 @@ class _BuildDesktopNavState extends State<_BuildDesktopNav>
   }
 }
 
-class _BuildMobileNav extends StatelessWidget {
+class _BuildMobileNav extends StatefulWidget {
   const _BuildMobileNav(
-      {this.selectedIndex, this.destinations, this.onItemTapped});
+      {this.selectedIndex, this.destinations, this.folders, this.onItemTapped});
   final int selectedIndex;
   final Map<String, String> destinations;
+  final Map<String, String> folders;
   final void Function(int) onItemTapped;
 
+  @override
+  __BuildMobileNavState createState() => __BuildMobileNavState();
+}
+
+class __BuildMobileNavState extends State<_BuildMobileNav>
+    with TickerProviderStateMixin {
+  final GlobalKey _bottomDrawerKey = GlobalKey(debugLabel: 'Bottom Drawer');
+  int _destinationsCount;
+  AnimationController _drawerController;
+  AnimationController _dropArrowController;
+  Map<String, int> _destinationsWithIndex;
+  String _currentDestination;
+
+  @override
+  void initState() {
+    super.initState();
+    _drawerController = AnimationController(
+      duration: const Duration(milliseconds: 350),
+      value: 0,
+      vsync: this,
+    )..addListener(() {
+        if (_drawerController.value < 0.01) {
+          setState(() {
+            //Reload state when drawer is at its smallest to toggle visibility
+            //If state is reloaded before this drawer closes abruptly instead
+            //of animating.
+          });
+        }
+      });
+
+    _dropArrowController = AnimationController(
+        duration: const Duration(milliseconds: 350), vsync: this)
+      ..addListener(() {});
```

Comment:
what is this listener for? Does not look like it is needed

Reply:
Thanks I missed this.